shared/src/main/java/com/faforever/neroxis/mask/BooleanMask.java

```diff
@@ -229,10 +229,13 @@ protected BooleanMask setSizeInternal(int newSize) {
             } else if (oldSize != newSize) {
                 long[] oldMask = mask;
                 initializeMask(newSize);
-                Map<Integer, Integer> coordinateMap = getSymmetricScalingCoordinateMap(oldSize, newSize);
-                applyWithSymmetry(SymmetryType.SPAWN, (x, y) -> {
-                    boolean value = getBit(coordinateMap.get(x), coordinateMap.get(y), oldSize, oldMask);
-                    applyAtSymmetryPoints(x, y, SymmetryType.SPAWN, (sx, sy) -> setPrimitive(sx, sy, value));
+
+                float scale = (float)oldSize / (float)newSize;
+
+                apply((x, y) -> {
+                    int sx = (int)(x * scale);
+                    int sy = (int)(y * scale);
+                    setPrimitive(x, y, getBit(sx, sy, oldSize, oldMask));
```

Comment:
One note here is that we still need the symmetricScalingCoordinateMap, otherwise  the map will end up with a one pixel shift in some parts.

This is because the conversion from float to int is not symmetric around the center of the map. Sof for example with a ratio of .5 11->5 while (100-11)->(50-5.5)->(44.5)->44 when really we would want it to go to 45 to have the same distance from the edge assuming a 100 pixel size map to a 50 pixel size map.

Reply:
Oh yep, I see what you mean.
I checked this scenario, scaling from 101 down to 50, and it was shifting 1 pixel to the left. Classic.
I guess to scale these ones properly, I'd have to scale from the middle of each pixel, ie add 0.5 then scale then subtrct 0.5

I've reverted this to use the getSymmetricScalingCoordinateMap() again.
But used the `apply()` function to ensure each pixel is scaled onto the new mask.

The `applyWithSymmetry()` has problems because it skips some pixels during scaling, and it sets some pixels twice:
So `applyWithSymmetry()` looks like this:
![image](https://github.com/user-attachments/assets/2841afe3-d173-455a-be43-4e1825dbcb29)